File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int[] countMap = new int[26];
        int index = -1;
        for (var c : s.toCharArray()) {
            countMap[c-'a']++;
        }
        for (int i = 0; i < s.length(); i++) {
            if (countMap[s.charAt(i)-'a'] == 1) return i;
        }
        return index;
    }
}

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.kt
class Solution {
    fun firstUniqChar(s: String): Int {
        val countMap = IntArray(26)
        for (c in s) {
            countMap[c-'a']++
        }
        for (i in s.indices) {
            if (countMap[s[i]-'a'] == 1) {
                return i
            }
        }
        return -1
    }
}